src/tools/clu_rand.c

```diff
@@ -0,0 +1,194 @@
+    int longIndex = 1;
+    WOLFSSL_BIO *bioOut = NULL;
+    byte *buf = NULL;
+
+    /* last parameter is the rand bytes output size */
+    if (XSTRNCMP("-h", argv[argc-1], 2) == 0) {
+        wolfCLU_RandHelp();
+        return WOLFCLU_SUCCESS;
+    }
+    else {
+        size = XATOI(argv[argc-1]);
+        if (size <= 0) {
+            WOLFCLU_LOG(WOLFCLU_L0, "unable to convert %s to a number",
+                    argv[argc-1]);
+            ret = WOLFCLU_FATAL_ERROR;
+        }
+    }
+
+    opterr = 0; /* do not display unrecognized options */
+    optind = 0; /* start at indent 0 */
+    while ((option = getopt_long_only(argc, argv, "",
+                   rand_options, &longIndex )) != -1) {
+        switch (option) {
+            case WOLFCLU_BASE64:
+                useBase64 = 1;
+                break;
+
+            case WOLFCLU_OUTFILE:
+                bioOut = wolfSSL_BIO_new_file(optarg, "wb");
+                if (bioOut == NULL) {
+                    WOLFCLU_LOG(WOLFCLU_L0, "unable to open output file %s",
+                            optarg);
+                    ret = WOLFCLU_FATAL_ERROR;
+                }
+                break;
+
+            case WOLFCLU_HELP:
+                wolfCLU_RandHelp();
+                return WOLFCLU_SUCCESS;
+
+            case ':':
+            case '?':
+                break;
+
+            default:
+                /* do nothing. */
+                (void)ret;
+        }
+    }
+
+
+    if (ret == WOLFCLU_SUCCESS) {
+        buf = (byte*)XMALLOC(size, HEAP_HINT, DYNAMIC_TYPE_RNG);
+        if (buf == NULL) {
+            ret = WOLFCLU_FATAL_ERROR;
+        }
+    }
+
+    if (ret == WOLFCLU_SUCCESS) {
+        WC_RNG rng;
+        if (wc_InitRng(&rng) != 0) {
+            WOLFCLU_LOG(WOLFCLU_L0, "unable to initialize RNG");
+            ret = WOLFCLU_FATAL_ERROR;
+        }
+        else {
+            if (wc_RNG_GenerateBlock(&rng, buf, size) != 0) {
+                WOLFCLU_LOG(WOLFCLU_L0, "unable to generate RNG block");
+                ret = WOLFCLU_FATAL_ERROR;
+            }
+            wc_FreeRng(&rng);
+        }
+    }
+
+    /* setup output bio to stdout if not set */
+    if (ret == WOLFCLU_SUCCESS && bioOut == NULL) {
+        bioOut = wolfSSL_BIO_new(wolfSSL_BIO_s_file());
+        if (bioOut == NULL) {
+            ret = WOLFCLU_FATAL_ERROR;
+        }
+        else {
+            if (wolfSSL_BIO_set_fp(bioOut, stdout, BIO_NOCLOSE)
+                    != WOLFSSL_SUCCESS) {
+                ret = WOLFCLU_FATAL_ERROR;
+            }
+        }
+    }
+
+    /* check and convert to base64 */
+    if (ret == WOLFCLU_SUCCESS && useBase64) {
+        byte *base64 = NULL;
+        word32 base64Sz;
+
+        if (Base64_Encode(buf, size, NULL, &base64Sz) != LENGTH_ONLY_E) {
+            WOLFCLU_LOG(WOLFCLU_L0, "error getting size for base64");
+            ret = WOLFCLU_FATAL_ERROR;
+        }
+
+        base64 = (byte*)XMALLOC(base64Sz, HEAP_HINT, DYNAMIC_TYPE_RNG);
```

Comment:
`DYNAMIC_TYPE_RNG` should change here, too. Corresponding `XFREE`s will need to change as well.